### DISCUSSION TITLE
Using markdownRendererFencedCodeAttributeContextBegin-End

### DISCUSSION THREAD

**Discussion by csrmy** [Witiko/markdown#481] [Q&A] [answered]
Hi,

I would like to reproduce the look of the markdown code blocks in Github like so:
`Inline code exemple`
```
void CodeBlockExemple(void)
{
    return;
}
```

I don´t know if it's the good approach but I planned on using `\colorbox` to get the result I wanted as so:\
`Inline \colorbox{red} {code} exemple` (red color just for exemple)
![image](https://github.com/user-attachments/assets/516e94a5-1150-4d1f-945f-83b2c17c3223)

And I am having a hard time using `markdownRendererFencedCodeAttributeContextBegin` and `markdownRendererFencedCodeAttributeContextEnd`. I checked in the documentation, there is an example of how to use it with the `minted` package but I am not getting where I want with the `\colorbox` command.

What could I do different ?
Thank you so much in advance for the time and help !

**Comment by Witiko** [comment, marked as answer]
> And I am having a hard time using `markdownRendererFencedCodeAttributeContextBegin` and `markdownRendererFencedCodeAttributeContextEnd`. I checked in the documentation, there is an example of how to use it with the `minted` package but I am not getting where I want with the `\colorbox` command.

[The fenced code attribute context renderers][2] are only produced when [the `fencedCodeAttributes` option][1] has been enabled and there are HTML attributes attached to a fenced code block.

For example, assume that you have the following Markdown document `example.md`:

``` md
~~~~ {#mycode .haskell .numberLines startFrom=100}
qsort []     = []
qsort (x:xs) = qsort (filter (< x) xs) ++ [x] ++
               qsort (filter (>= x) xs)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Processing the document `example.md` with the Markdown package produces the following TeX output:

``` sh
$ markdown-cli fencedCodeAttributes=true example.md
```
``` tex
\markdownRendererDocumentBegin
\markdownRendererFencedCodeAttributeContextBegin\markdownRendererAttributeIdentifier{mycode}\markdownRendererAttributeClassName{haskell}\markdownRendererAttributeClassName{numberLines}\markdownRendererAttributeKeyValue{startFrom}{100}\markdownRendererInputFencedCode{./6c80369efc4b7e2154ad1c5b0019af21.verbatim}{}{}\markdownRendererFencedCodeAttributeContextEnd{}\markdownRendererDocumentEnd
```

Unless you wish to use HTML attributes, I would redefine `\markdownRendererInputFencedCode` directly to apply any colors you want.

P.S.: In the future, please address questions that aren't bug reports to https://github.com/Witiko/markdown/discussions.

 [1]: https://witiko.github.io/markdown/#option-fencedcodeattributes
 [2]: https://witiko.github.io/markdown/#fenced-code-attribute-context-renderers

**Reply by csrmy** [reply]
Thank you for your reply !

I have setup my `.tex` file with this:
```
\global\def\markdownRendererInputVerbatim#1{%
     \verbatiminput{#1}%
}

\global\def\markdownRendererInputFencedCode#1#2#3{%
	\markdownRendererInputVerbatim{#1}%
}
```
That reproduce the default rendering of fenced code blocks. But everytime I try to style the verbatim input I get this error:
`! LaTeX Error: Something's wrong--perhaps a missing \item`

I suppose the logic is that the fenced code blocks are stored in temporary files during the markdown parsing and we render those files with `\verbatiminput`. So it'a problem of styling the rendering of verbatiminput rather than a markdown rendering problem. I will look into it.

Thank you again for your help and time, it's very nice !
